vispy/visuals/isoline.py

```diff
@@ -0,0 +1,190 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2014, Vispy Development Team.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+
+from __future__ import division
+
+import numpy as np
+
+from .line import LineVisual
+from ..color import ColorArray
+from ..ext.six import string_types
+from ..color.colormap import _normalize, get_colormap
+
+
+def iso_mesh_line(vertices, tris, vertex_data, levels):
+    """Generate an isocurve from vertex data in a surface mesh.
+
+    Parameters
+    ----------
+    vertices : ndarray, shape (Nv, 3)
+        Vertex coordinates.
+    tris : ndarray, shape (Nf, 3)
+        Indices of triangular element into the vertices array.
+    vertex_data : ndarray, shape (Nv,)
+        data at vertex.
+    levels : ndarray, shape (Nl,)
+        Levels at which to generate an isocurve
+
+    Returns
+    -------
+    lines : ndarray, shape (Nvout, 3)
+        Vertex coordinates for lines points
+    connects : ndarray, shape (Ne, 2)
+        Indices of line element into the vertex array.
+    vertex_level: ndarray, shape (Nvout,)
+        level for vertex in lines
+ 
+    Notes
+    -----
+    Uses a marching squares algorithm to generate the isolines.
+    """
+
+    lines = None
+    connects = None
+    vertex_level = None
+    if all([isinstance(x, np.ndarray) for x in (vertices, tris,
+            vertex_data, levels)]):
```

Comment:
What is the purpose of this check? If this evaluates to `False`, `None` is returned for every variable. Maybe instead you want to say `if not all(...) raise ValueError('all inputs must be numpy arrays')` and then un-indent all of the following lines...?